parsely/src/main/java/com/parsely/parselyandroid/UpdateEngagementIntervalCalculator.java

```diff
@@ -0,0 +1,23 @@
+package com.parsely.parselyandroid;
+
+import androidx.annotation.NonNull;
+
+import java.util.Calendar;
+import java.util.TimeZone;
+
+class UpdateEngagementIntervalCalculator {
+
+    private static final long MAX_TIME_BETWEEN_HEARTBEATS = 60 * 60;
+    private static final long OFFSET_MATCHING_BASE_INTERVAL = 35;
+    private static final double BACKOFF_PROPORTION = 0.3;
+
+    long updateLatestInterval(@NonNull final Calendar startTime) {
+        Calendar now = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+        long totalTrackedTime = (now.getTime().getTime() - startTime.getTime().getTime()) / 1000;
+        double totalWithOffset = totalTrackedTime + OFFSET_MATCHING_BASE_INTERVAL;
+        double newInterval = totalWithOffset * BACKOFF_PROPORTION;
+        long clampedNewInterval = (long) Math.min(MAX_TIME_BETWEEN_HEARTBEATS, newInterval);
+        System.out.println("New interval: " + clampedNewInterval*1000);
```

Comment:
1. Warning (⚠️): I am not sure if this was done due to an oversight or by design but usually `ParselyTracker.PLog(...)` is used instead. Consider either removing this completely or using `ParselyTracker.PLog(...)` instead.
2. Question (❓): Now that I am seeing the `System.out.println(...)` logging, what about replacing the whole `ParselyTracker.PLog(...)` with the native Android logger, of course, and this way to be having a more fine-grained level of logging (`Log.d/i/e(...)`), as a separate issue/PR, wdyt? 🤔

Reply:
Thanks, I'll remove this log.

> System.out.println(...) logging, what about replacing the whole ParselyTracker.PLog(...) with the native Android logger,

We could do this, but at the same time, I don't see really a value in such change for this SDK. Could you elaborate, what would we gain by using the Android logger?

Reply:
> Could you elaborate, what would we gain by using the Android logger?

Well, as I said above, a more fine-grained level of logging (`Log.d/i/e(...)`). Apart from that, I am not sure sure if using `System.out.println(...)` logging has some performance implication for apps that are using this SDK, or even if apps can remove this logging if they want to, with the native Android logging, I think the can, right... 🤷 🤔

Reply:
Gotcha, thanks. I've added an issue for that: https://github.com/Parsely/parsely-android/issues/83 - I thought that `PLog` logs only if SDK has `debug` flag to `true`.

Reply:
Awesome, thanks for opening this issue @wzieba ! 🙇 

> I thought that `PLog` logs only if SDK has `debug` flag to `true`.

Wishful thinking... 😅 🤷